spec-tests/src/test/java/org/candlepin/spec/InactiveConsumerCleanerJobTest.java

```diff
@@ -0,0 +1,124 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.ConsumerTypeDTO;
+import org.candlepin.dto.api.v1.NestedOwnerDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.spec.bootstrap.client.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.ConsumerTypeClient;
+import org.candlepin.spec.bootstrap.client.JobsClient;
+import org.candlepin.spec.bootstrap.client.OwnerClient;
+import org.candlepin.spec.bootstrap.client.SpecTestFixture;
+import org.candlepin.spec.bootstrap.data.ConsumerDTOBuilder;
+import org.candlepin.spec.bootstrap.data.OwnerDTOBuilder;
+import org.candlepin.spec.bootstrap.data.Util;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class InactiveConsumerCleanerJobTest extends SpecTestFixture {
+
+    private static final String JOB_KEY = "InactiveConsumerCleanerJob";
+    private static final int DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS = 397;
+
+    private ConsumerClient consumerClient;
+    private OwnerClient ownerClient;
+    private JobsClient jobsClient;
+    private ConsumerTypeClient consumerTypeClient;
+
+    private OwnerDTO owner;
+    private ConsumerTypeDTO consumerType;
+
+    @BeforeEach
+    private void beforeEach() throws ApiException {
+        consumerClient = getClient(ConsumerClient.class);
+        ownerClient = getClient(OwnerClient.class);
+        jobsClient = getClient(JobsClient.class);
+        consumerTypeClient = getClient(ConsumerTypeClient.class);
+
+        if (owner == null) {
+            owner = ownerClient.createOwner(new OwnerDTOBuilder().build());
+        }
+
+        if (consumerType == null) {
+            ConsumerTypeDTO newType = new ConsumerTypeDTO();
+            newType.setManifest(false);
+            newType.setLabel(Util.randomString("Test Type"));
+            consumerType = consumerTypeClient.createConsumerType(newType);
+        }
+    }
+
+    @Test
+    public void shouldDeleteInactiveConsumers() throws Exception {
+        Instant inactiveTime = Instant.now().minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS + 10,
+            ChronoUnit.DAYS);
+        Instant activeTime = Instant.now().minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS - 10,
+            ChronoUnit.DAYS);
+
+        ConsumerDTO inactiveConsumer = buildConsumer(inactiveTime.atOffset(ZoneOffset.UTC));
+        ConsumerDTO activeConsumer = buildConsumer(activeTime.atOffset(ZoneOffset.UTC));
+        ConsumerDTO activeConsumer2 = buildConsumer(activeTime.atOffset(ZoneOffset.UTC));
```

Comment:
Let's make one of the consumers have a null lastcheckin time. Their updated date will not be more than 30 days in the past, so we can assert that they will not get deleted.
Adding a manifest consumer that is inactive should also be easy (and assert they're not getting deleted)

Reply:
Also, we really need to verify that inactive consumers that have entitlements are not getting deleted, but I understand that would mean porting over the pools, products and entitlements client objects to java, so it is probably too much for a single task. 

So this begs the question: Maybe we should keep writing spec tests in Ruby for now (at least the complicated ones like this one), until all those are migrated to Java?
What does everyone think? @Januson @Ceiu @wottop @JoshAlbrecht23

Reply:
I'm torn. While it may be out of scope for this particular PR, doing it in Ruby at all is just doing work that we know is going to be ripped out in the immediate future. If it's just a matter of porting over client objects, that should be pretty simple, right? Maybe that could be a separate PR that blocks this one and then this gets rebased on it.

If it's a complicated task to get the requirements for writing a Java spec test in, then I could see defaulting to Ruby for the time being. So I guess, for me, it comes down to how much work it'd be to spin up the necessary components to write the desired tests.

Reply:
I like opening up a separate PR for porting over the client objects and then rebasing this PR with those changes! I'm working on that now.

Reply:
Created the following PR for adding the client objects: https://github.com/candlepin/candlepin/pull/3407/

Reply:
Closed the PR as we can use the *API.java classes and only create *Client.java classes for rare cases.